front/lib/api/data_sources.ts

```diff
@@ -382,14 +382,6 @@ export async function upsertDocument({
   const documentId = name;
   const documentParents = parents || [];
 
-  // Ensure that the documentId is included in the parents as the first item.
-  // remove it if it's already present and add it as the first item.
-  const indexOfDocumentId = documentParents.indexOf(documentId);
-  if (indexOfDocumentId !== -1) {
-    documentParents.splice(indexOfDocumentId, 1);
-  }
-  documentParents.unshift(documentId);
```

Comment:
should we return an `Err` to crash more directly?

Reply:
We know we have cases where we get into this (first entry in parents is not the document id) - especially if documentId is not in the same format as parents , so we should not crash until we're sure we sending parents[] with same format as documentId

Reply:
IMO we should just not do this, or maybe only if parents is null or empty array. @Fraggle I think you added this , do you remember in which case it was used ?

Reply:
Yes extremely hard to understand a system that mutates what you give it. The system should check / enforce not modify

Reply:
Now the question is who wrote that and are they relying on it in any way?

Reply:
I was adding it to ensure that parent was the first id, you can remove the code until it's fully expected.

Reply:
BTW, we had a [discussion](https://github.com/dust-tt/dust/pull/8716#discussion_r1846621472), this method is not called by connectors (unless i'm mistaken), only by front when we are uploading to folders.

Reply:
right, not this one, but the handleDataSourceTableCSVUpsert below is used by connectors upsert (through v1 api) . We should also remove the code there

Reply:
As per half-IRL half-discussion here: we keep these 2 for now but remove in `handleDataSourceTableCSVUpsert`